File: erp-customer-service/src/main/java/com/crossborder/erp/customer/entity/Customer.java
package com.crossborder.erp.customer.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.annotation.FieldFill;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 客户实体 - 跨境电商客户管理
 */
@Data
@TableName("t_customer")
public class Customer {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 客户姓名
     */
    private String name;

    /**
     * 客户邮箱
     */
    private String email;

    /**
     * 联系电话
     */
    private String phone;

    /**
     * 客户来源平台
     */
    private String platform;

    /**
     * 平台客户ID
     */
    private String platformCustomerId;

    /**
     * 国家/地区
     */
    private String country;

    /**
     * 省份/州
     */
    private String state;

    /**
     * 城市
     */
    private String city;

    /**
     * 详细地址
     */
    private String address;

    /**
     * 邮政编码
     */
    private String postalCode;

    /**
     * 客户等级（VIP/NORMAL/NEW）
     */
    private String level;

    /**
     * 订单总数
     */
    private Integer totalOrders;

    /**
     * 累计消费金额
     */
    private BigDecimal totalSpent;

    /**
     * 平均订单金额
     */
    private BigDecimal avgOrderValue;

    /**
     * 最后下单时间
     */
    private LocalDateTime lastOrderAt;

    /**
     * 客户备注
     */
    private String remark;

    /**
     * 状态（ACTIVE/INACTIVE/BLOCKED）
     */
    private String status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: erp-web/src/api/customer.ts
import axios from 'axios'
import { ElMessage } from 'element-plus'

const request = axios.create({
  baseURL: '/api',
  timeout: 10000
})

request.interceptors.response.use(
  response => response.data,
  error => {
    ElMessage.error(error.message || '请求失败')
    return Promise.reject(error)
  }
)

export interface Customer {
  id: number
  name: string
  email: string
  phone: string
  platform: string
  platformCustomerId: string
  country: string
  state: string
  city: string
  address: string
  postalCode: string
  level: string
  totalOrders: number
  totalSpent: number
  avgOrderValue: number
  lastOrderAt: string
  remark: string
  status: string
  createTime: string
  updateTime: string
}

export interface CustomerStats {
  totalCustomers: number
  vipCount: number
  activeCount: number
  totalRevenue: number
  avgSpent: number
}

export interface PageResult<T> {
  current: number
  size: number
  total: number
  records: T[]
}

export const customerApi = {
  // 创建客户
  create: (data: Customer) => 
    request.post<any, number>('/customer/customer/create', data),

  // 获取客户详情
  getById: (id: number) => 
    request.get<any, Customer>(`/customer/customer/${id}`),

  // 分页查询客户
  getList: (params: {
    current: number
    size: number
    keyword?: string
    platform?: string
    level?: string
    country?: string
  }) => request.get<any, PageResult<Customer>>('/customer/customer/list', { params }),

  // 获取客户统计
  getStats: () => 
    request.get<any, CustomerStats>('/customer/customer/stats'),

  // 获取VIP客户
  getVipList: () => 
    request.get<any, Customer[]>('/customer/customer/vip'),

  // 更新客户
  update: (id: number, data: Customer) => 
    request.put<any, void>(`/customer/customer/${id}`, data),

  // 删除客户
  delete: (id: number) => 
    request.delete<any, void>(`/customer/customer/${id}`)
}